src/ethereum/frontier/state.py

```diff
@@ -0,0 +1,218 @@
+
+
+@dataclass
+class State:
+    """
+    Contains all information that is preserved between transactions.
+    """
+
+    _main_trie: Trie[Account] = field(
+        default_factory=lambda: Trie(secured=True, default=EMPTY_ACCOUNT)
+    )
+    _storage_tries: Dict[Address, Trie[U256]] = field(default_factory=dict)
+
+
+def get_account(state: State, address: Address) -> Account:
+    """
+    Get the `Account` object at an address. Returns EMPTY_ACCOUNT if there
+    is no account the address.
+
+    Parameters
+    ----------
+    state: `State`
+        The state
+    address : `Address`
+        Address to lookup.
+
+    Returns
+    -------
+    account : `Account`
+        Account at address.
+    """
+    account = trie_get(state._main_trie, address)
+    assert isinstance(account, Account)
+    return account
+
+
+def set_account(state: State, address: Address, account: Account) -> None:
+    """
+    Set the `Account` object at an address.
+
+    Parameters
+    ----------
+    state: `State`
+        The state
+    address : `Address`
+        Address to set.
+    account : `Account`
+        Account to set at address.
+    """
+    trie_set(state._main_trie, address, account)
+
+
+def destroy_account(state: State, address: Address) -> None:
+    """
+    Completely remove the account at `address` and all of its storage.
+
+    This function is made available exclusively for the `SELFDESTRUCT`
+    opcode. It is expected that `SELFDESTRUCT` will be disabled in a future
+    hardfork and this function will be removed.
+
+    Parameters
+    ----------
+    state: `State`
+        The state
+    address : `Address`
+        Address of account to destroy.
+    """
+    del state._storage_tries[address]
+    set_account(state, address, EMPTY_ACCOUNT)
+
+
+def get_storage(state: State, address: Address, key: Bytes) -> U256:
+    """
+    Get a value at a storage key on an account.
+
+    Parameters
+    ----------
+    state: `State`
+        The state
+    address : `Address`
+        Address of the account.
+    key : `Bytes`
+        Key to lookup.
+
+    Returns
+    -------
+    value : `U256`
+        Value at the key.
+    """
+    trie = state._storage_tries.get(address)
+    if trie is None:
+        return U256(0)
+
+    value = trie_get(trie, key)
+
+    assert isinstance(value, U256)
+    return value
+
+
+def set_storage(
+    state: State, address: Address, key: Bytes, value: U256
+) -> None:
+    """
+    Set a value at a storage key on an account.
+
+    Parameters
+    ----------
+    state: `State`
+        The state
+    address : `Address`
+        Address of the account.
+    key : `Bytes`
+        Key to set.
+    value : `U256`
+        Value to set at the key.
+    """
+    trie = state._storage_tries.get(address)
+    if trie is None:
+        trie = Trie(secured=True, default=U256(0))
+        state._storage_tries[address] = trie
+    trie_set(trie, key, value)
```

Comment:
I feel it would be giving more clarity to the readers if we mentioned here and in docstring that if the value is 0, and if the key is present in the storage, then it would be deleted from the storage. Please let me know what you think. Thanks.